packages/rs-dapi-client/src/address_list.rs

```diff
@@ -40,12 +74,19 @@ impl Address {
     }
 }
 
+/// [AddressList] errors
+#[derive(Debug, thiserror::Error)]
+pub enum AddressListError {
+    #[error("address {0} not found in the list")]
+    AddressNotFound(Uri),
+}
+
 /// A structure to manage peer's addresses to select from
 /// for [DapiRequest](crate::DapiRequest) execution.
 #[derive(Debug)]
 pub struct AddressList {
-    addresses: Vec<Address>,
-    base_ban_period: time::Duration,
+    addresses: HashSet<Address>,
```

Comment:
I explained on a call why a HashMap was better. I will do it again today. For a Hashset, if you knew only the URI you would need to create a new address for it, which is very counter intuitive. For example if I wanted to ban 111.2.4.5 from the address list, I would need to create a new address then ban it. Seems wrong to me.

Reply:
If you REALLY prefer a Hashset, you could get by by renaming ban and clear_ban to ban_address and clean_ban_address, and create new methods for ban and clear ban that take a URI

Reply:
Renamed and closed these methods for the crate only